wellcomecollection/wellcomecollection.org
# Re-add svgo when compiling icons

## What does this change?
For reasons I can't explain I apparently removed the svgo plugin (a tool for optimising svgs) in a [PR in 2019](https://github.com/wellcomecollection/wellcomecollection.org/pull/4346). Possibly because it was removing the `viewbox` attribute which we rely on, but rather than figuring out the part of the config I had to change [I turned it off altogether](https://github.com/wellcomecollection/wellcomecollection.org/pull/4346/files#diff-e8c293f1972c360d9ae1b630128e7ce27fe531380f3bce400080ac3a90110f7eR6) (`--no-svgo`). I have no idea if this is what happened (I don't remember doing this at all), but it doesn't seem beyond the realm of possibility.

@rcantin-w pointed out that we don't want to allow `fill` attributes that aren't `currentColor` and this made me wonder whether that was something possible with svgo, which then made me realise we had it turned off.

Using JS instead of YAML because I couldn't figure out how to add attributes (neither using the commented out example in the file, or the SO suggestions I found). The SVGO docs are expressed in JS so made sense to use JS.

## How to test
- Run Cardigan and check the [icons](http://localhost:9001/?path=/story/global-icons--page) render
- Verify icons that should change colour to match `currentColor` do so (e.g. [ButtonSolid](http://localhost:9001/?path=/story/components-buttons-basics--button-solid))

## How can we measure success?
- Smaller icons
- No incorrect/unnecessary attributes making it through from the svgs to the React components

## Have we considered potential risks?
The optimisations make the icons look different/worse somehow, although I don't believe this has happened through checking cardigan
